tests/ui/fn_to_numeric_cast_with_truncation.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(tool_lints)]
+
+#![warn(clippy::fn_to_numeric_cast_with_truncation)]
+#![allow(clippy::fn_to_numeric_cast)]
+
+fn foo() -> String { String::new() }
+
+fn test_fn_to_numeric_cast_with_truncation() {
+    let _ = foo as i8;
+    let _ = foo as i16;
+    let _ = foo as i32;
+    let _ = foo as u8;
+    let _ = foo as u16;
+    let _ = foo as u32;
+
+    // TODO: Is it bad to have these tests?
```

Comment:
There are compile-test flags you can set to ignore tests on specific architectures.